solr-search/src/main/java/edu/unc/lib/dl/search/solr/model/IndexDocumentBean.java

```diff
@@ -15,503 +15,481 @@
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.RESOURCE_TYPE;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.RESOURCE_TYPE_SORT;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.ROLE_GROUP;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.ROLLUP_ID;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.STATUS;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.STATUS_TAGS;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.SUBJECT;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.TIMESTAMP;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.TITLE;
+import static edu.unc.lib.dl.search.solr.util.SearchFieldKeys.VERSION;
+
 import java.text.ParseException;
 import java.util.Date;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 
 import org.apache.solr.client.solrj.beans.Field;
 
+import edu.unc.lib.dl.fcrepo4.PIDs;
 import edu.unc.lib.dl.fedora.PID;
 import edu.unc.lib.dl.util.DateTimeUtil;
 
+
 /**
  *
- * @author bbpennel, harring
+ * @author bbpennel
+ * @author harring
  *
  */
+@SuppressWarnings("unchecked")
 public class IndexDocumentBean {
     protected PID pid;
-    protected String id;
-    protected List<String> ancestorPath;
-    protected String ancestorIds;
-    protected String parentCollection;
-    protected String parentUnit;
-    protected String label;
-
-    protected List<String> scope;
-    protected String rollup;
-    protected Boolean isPart;
-    protected Long _version_;
-
-    protected List<String> datastream;
-    protected Long filesizeSort;
-    protected Long filesizeTotal;
-
-    protected List<String> relations;
-
-    protected List<String> contentModel;
-    protected String resourceType;
-    protected Integer resourceTypeSort;
-
-    protected String creatorSort;
-    protected String defaultSortType;
-    protected Long displayOrder;
-
-    protected List<String> contentType;
-
-    protected Date timestamp;
-    protected Date lastIndexed;
-
-    protected List<String> roleGroup;
-    protected List<String> readGroup;
-    protected List<String> adminGroup;
-    protected List<String> status;
-    protected List<String> contentStatus;
-    protected HashSet<String> statusTags;
-
-    protected List<String> identifier;
-    protected String identifierSort;
-
-    // Descriptive fields
-    protected String title;
-    protected List<String> otherTitle;
-    protected String abstractText;
-    protected List<String> keyword;
-    protected List<String> subject;
-    protected List<String> language;
-    protected List<String> creator;
-    protected List<String> contributor;
-    protected List<String> department;
-
-    protected Date dateCreated;
-    protected Date dateAdded;
-    protected Date dateUpdated;
-
-    protected String citation;
-
-    protected String fullText;
+
+    private Map<String, Object> fields;
```

Comment:
Oh yeah, I guess this is why I started calling the getters everywhere instead of referencing variables